common/diagnostics/StatisticReporter.java

```diff
@@ -10,49 +10,117 @@
 import org.slf4j.LoggerFactory;
 
 import javax.net.ssl.HttpsURLConnection;
+import java.io.IOException;
+import java.net.ProtocolException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 
-import static java.util.concurrent.TimeUnit.HOURS;
+import static java.lang.Math.abs;
+import static java.util.concurrent.TimeUnit.MINUTES;
 
 public class StatisticReporter {
-    protected static final Logger LOG = LoggerFactory.getLogger(StatisticReporter.class);
+    // Modify calculateInitialDelay() if you change this value!
```

Comment:
I didn't come up with a good formula of `calculateInitialDelay` (that calculates at what minute of the current or the future hour this server needs to send the first snapshot of diagnostics to be in sync with other servers of the deployment) based on different intervals, so I hardcoded it considering that it should be 60 now.